gtk4-macros/src/attribute_parser.rs

```diff
@@ -273,9 +273,28 @@ pub fn parse_fields(fields: &Fields) -> Result<Vec<AttributedField>, Error> {
     let mut attributed_fields = Vec::new();
 
     for field in fields {
+        let mut has_attr = false;
         if !field.attrs.is_empty() {
             if let Some(attributed_field) = parse_field(field)? {
-                attributed_fields.push(attributed_field)
+                attributed_fields.push(attributed_field);
+                has_attr = true;
+            }
+        }
+        if !has_attr {
+            if let syn::Type::Path(syn::TypePath { path, .. }) = &field.ty {
+                if path.leading_colon.is_none()
+                    && path.segments.len() == 1
+                    && matches!(
+                        &path.segments[0].arguments,
+                        syn::PathArguments::AngleBracketed(_)
+                    )
+                    && path.segments[0].ident == "TemplateChild"
+                {
+                    proc_macro_error::emit_warning!(
+                        field,
+                        "field with type `TemplateChild` possibly missing #[template_child] attribute",
+                    );
```

Comment:
Maybe this should also get a compile test?

Reply:
We would need to have a way to only run the test on nightly, I don't believe there is an easy way to do that

Reply:
Just changed it to an error and added a compile test

Reply:
Makes sense